api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -462,10 +469,17 @@ public BiometricData saveBiometricsForPatient(Patient patient, BiometricData bio
 					log.warn("Found " + patients.size() + " patients with the same biometric id: " + subjectId);
 				}
 				for (Patient patient : patients) {
+					possessNationalFpId = possessNationalFpId(patient) || possessNationalFpId;
```

Comment:
This line looks quite confusing. I think we should avoid the same parameter and method names.
Moreover, if it is possible we can use Java 8 stream with findAny() method which allows to determine if any element possess national fp id with more 'clean code' way

Reply:
In this module we don't have Java 8

Reply:
Ok, however parameter name and method name shouldn't be the same